// react-native-form-builder/src/fields/form/index.js
import PropTypes from 'prop-types';
import React, { Component } from 'react';
import { View, Text } from 'native-base';
import { TextInput } from 'react-native';
import GenerateForm from '../../formBuilder';

export default class FormField extends Component {
  static propTypes = {
    attributes: PropTypes.object,
    theme: PropTypes.object,
    updateValue: PropTypes.func,
    autoValidation: PropTypes.bool,
    customValidation: PropTypes.func,
    customComponents: PropTypes.object,
  }
  constructor(props) {
    super(props);
    this.onValueChange = this.onValueChange.bind(this);
  }
  componentDidMount() {
    this.props.updateValue(this.props.attributes.name, this.group.getValues());
  }
  onValueChange() {
    this.props.updateValue(this.props.attributes.name, this.group.getValues());
  }
  handleChange(text) {
    this.setState({
      value: text,
    }, () => this.props.updateValue(this.props.attributes.name, text));
  }
  ShowMaxAlert = (EnteredValue) =>{
      var TextLength = EnteredValue.length.toString() ;
      if(TextLength == 140){
        Alert.alert("Sorry, you've reached the max input limit.")
        // Put your code here which you want to execute when TextInput entered text reached to 10.
      }
  }
  render() {
    const {
      attributes,
      theme,
      autoValidation,
      customValidation,
      customComponents,
    } = this.props;
    return (
      <View>
        <View style={{ marginLeft: 20, paddingHorizontal: 0, paddingVertical: 0 }}>
          <TextInput onChangeText={ EnteredValue => this.ShowMaxAlert(EnteredValue) } maxLength={140} numberOfLines={1} style={{ color:'black', fontWeight: '500', fontSize: 17 }}>{attributes.label}</TextInput>
        </View>
        <View style={{ backgroundColor: 'transparent', paddingHorizontal: 0 }}>
          <GenerateForm
            ref={(c) => { this.group = c; }}
            onValueChange={this.onValueChange}
            autoValidation={autoValidation}
            customValidation={customValidation}
            customComponents={customComponents}
            showErrors
            fields={attributes.fields}
            theme={theme}
          />
        </View>
      </View>
    );
  }
}
